src/main/java/net/imglib2/roi/labeling/Labelings.java

```diff
@@ -118,4 +119,54 @@
 		resultMapping.setLabelSets( resultLabelSets );
 		return result;
 	}
+
+	/**
+	 * Return a {@link Set} of occurring pixel values in the {@link ImgLabeling} index image.
+	 *
+	 * @param img
+	 * 		Image labeling from which to extract the occurring pixel values
+	 * @param <T>
+	 * 		The type of labels assigned to pixels
+	 * @param <I>
+	 * 		The pixel type of the backing image
+	 *
+	 * @return {@link Set} of occurring pixel values
+	 */
+	static < T, I extends IntegerType< I > > Set< I > getOccurringPixelSets( ImgLabeling< T, I > img )
+	{
+		Set< I > occurringValues = new HashSet<>();
+		for ( I pixel : Views.iterable( img.getIndexImg() ) )
+		{
+			if ( pixel.getInteger() > 0 )
+				occurringValues.add( pixel.copy() );
```

Comment:
This seems unnecessarily expensive. The `pixel.copy()` happens every pixel, although the `occurringValues.add(...)` then will only actually add it to the set for a very small fraction of pixels.

A fix would be to first check `occurringValues.contains(pixel)`, then only do the copy if it's not contained yet.

Or use a Trove IntHashSet or something like that and work with the int values

Reply:
I didn't look at the memory usage (which might be better with a IntHashSet or HashSet<Integer>). In general I would prefer not to use too specialised of data structure (such as TIntHashSet).

I performed a simple speed test (1000x1000x1000 image, random pixel values, 1000 runs measured with imglib2 StopWatch). Here's what I get for a single test with fixed seed between all pixel range:

| Pixel values  | Original  | + checks |  TIntHash  |
| ------------- | ------------- | ------------- |------------- |
| 0-1 | 881.895 µs | 700.458 µs  | 805.410 µs |
| 0-64 | 910.525 µs | 802.697 µs  | 718.706 µs |
| 0-2048 | 975.518 µs | 764.218 µs | 742.928 µs |
| 0-65535 | 917.941 µs | 905.915 µs | 901.384 µs|

Values fluctuate of course if I run it again, so the difference in performance is not that clear. I added a commit with a `Set.contains` check in the if clause.

Reply:
Yes, I would also have expected more significant improvement. (Maybe the JIT could eliminate the object creation in the original code?!)